tests/test_config.py

```diff
@@ -0,0 +1,330 @@
+import sys
+from contextlib import nullcontext as does_not_raise
+from inspect import signature
+from typing import Any, ContextManager, Iterable, NamedTuple, Type, Union
+
+from dirty_equals import HasRepr
+
+from pydantic import BaseConfig, BaseModel, Extra, Field, PrivateAttr, PydanticSchemaGenerationError, ValidationError
+from pydantic.config import ConfigDict, _default_config
+from pydantic.errors import PydanticUserError
+
+if sys.version_info < (3, 9):
+    from typing_extensions import Annotated
+else:
+    from typing import Annotated
+
+import pytest
+
+
+@pytest.fixture(scope='session', name='BaseConfigModelWithStrictConfig')
+def model_with_strict_config():
+    class ModelWithStrictConfig(BaseModel):
+        a: int
+        # strict=False overrides the Config
+        b: Annotated[int, Field(strict=False)]
+        # strict=None or not including it is equivalent
+        # lets this field be overridden by the Config
+        c: Annotated[int, Field(strict=None)]
+        d: Annotated[int, Field()]
+
+        class Config:
+            strict = True
+
+    return ModelWithStrictConfig
+
+
+def _equals(a: Union[str, Iterable[str]], b: Union[str, Iterable[str]]) -> bool:
+    """
+    Compare strings with spaces removed
+    """
+    if isinstance(a, str) and isinstance(b, str):
+        return a.replace(' ', '') == b.replace(' ', '')
+    elif isinstance(a, Iterable) and isinstance(b, Iterable):
+        return all(_equals(a_, b_) for a_, b_ in zip(a, b))
+    else:
+        raise TypeError(f'arguments must be both strings or both lists, not {type(a)}, {type(b)}')
+
+
+@pytest.mark.filterwarnings('ignore:.* is deprecated.*:DeprecationWarning')
+class TestsBaseConfig:
+    def test_base_config_equality_defaults_of_config_dict_class(self):
+        for key, value in _default_config.items():
+            assert getattr(BaseConfig, key) == value
+
+    def test_config_and_module_config_cannot_be_used_together(self):
+        with pytest.raises(PydanticUserError):
+
+            class MyModel(BaseModel):
+                model_config = ConfigDict(title='MyTitle')
+
+                class Config:
+                    title = 'MyTitleConfig'
+
+    def test_base_config_properly_converted_to_dict(self):
+        class MyConfig(BaseConfig):
+            title = 'MyTitle'
+            frozen = True
+
+        class MyBaseModel(BaseModel):
+            class Config(MyConfig):
+                ...
+
+        class MyModel(MyBaseModel):
+            ...
+
+        expected = _default_config.copy()
+        expected['title'] = 'MyTitle'
+        expected['frozen'] = True
+        for k, v in expected.items():
+            assert MyModel.model_config[k] == v
```

Comment:
@samuelcolvin prior to my changes, this was:
```python
        expected = _default_config.copy()
        expected['title'] = 'MyTitle'
        expected['frozen'] = True
        assert MyModel.model_config == expected  # <-- affected line
```
Is the `.model_config` attribute supposed to hold all fields from `_default_config`? Considering `ConfigDict` has `total=False` I'm assuming it shouldn't need to have all fields present, but I wanted to make sure I understood the intent correctly.

Reply:
That was my proposal - `MyModel.model_config` is really a partial of `ConfigDict`, but has special logic if the value is missing.